### PR TITLE
Change link method of Vestige module

### LINKED ISSUE #2628: Vestige module is not listed despite it builds successfully
I'm using arch linux x64. I'm trying to get lmms version with vestige module, but it doesn't show up.
I have all deps installed, when running cmake, it states that both vst instrument and vst effect are ok, also I see that everything related to vst compiled and linked successfully, but when I'm opening lmms itself, vestige module is not listed on the list. Compiling with gcc 5.3.0.

**Edit: @tresf from conversation on `#lmms` after adding `qDebug()`**

```
"Cannot load library /home/tricobalt/lmms/build/plugins/libvestige.so: (../libvstbase.so: cannot open shared object file: No such file or directory)"
"Cannot load library /home/tricobalt/lmms/build/plugins/libvsteffect.so: (../libvstbase.so: cannot open shared object file: No such file or directory)"
```

### PR DESCRIPTION
**Pull request by liushuyu**
Try to fix #2628
At least this work (in fact just fixes #2628) for me (Arch Linux 64-bit)

[Review on Reviewable](https://reviewable.io/reviews/lmms/lmms/2632)

### CODE REVIEW

**Comment by Umcaruje** [comment]
Hi, the build fails for windows: 
```
CMake Error at cmake/modules/BuildPlugin.cmake:61 (INSTALL):
  install Library TARGETS given no DESTINATION!
Call Stack (most recent call first):
  plugins/vestige/CMakeLists.txt:6 (BUILD_PLUGIN)
CMake Error at cmake/modules/BuildPlugin.cmake:61 (INSTALL):
  install Library TARGETS given no DESTINATION!
Call Stack (most recent call first):
  plugins/vst_base/CMakeLists.txt:27 (BUILD_PLUGIN)

```

**Comment by liushuyu** [comment]
Yes, I noticed that... I made some fixes...

**Comment by liushuyu** [comment]
Finally ,it passed

File: plugins/vestige/CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,11 @@ IF(LMMS_SUPPORT_VST)
    INCLUDE_DIRECTORIES("${CMAKE_CURRENT_SOURCE_DIR}/../vst_base")
    LINK_DIRECTORIES("${CMAKE_CURRENT_BINARY_DIR}/../vst_base")
    LINK_LIBRARIES(vstbase)
-   BUILD_PLUGIN(vestige vestige.cpp vestige.h MOCFILES vestige.h EMBEDDED_RESOURCES "${CMAKE_CURRENT_SOURCE_DIR}/*.png")
+   IF(LMMS_BUILD_WIN32 OR LMMS_BUILD_WIN64 OR LMMS_BUILD_MSYS)
```
Comment (tresf) [inline plugins/vestige/CMakeLists.txt]: `LMMS_BUILD_WIN32` will do the trick.  The other two imply `LMMS_BUILD_WIN32`

File: plugins/vestige/CMakeLists.txt [hunk 2]
```diff
@@ -3,6 +3,11 @@ IF(LMMS_SUPPORT_VST)
...
+	BUILD_PLUGIN(vestige vestige.cpp vestige.h MOCFILES vestige.h EMBEDDED_RESOURCES "${CMAKE_CURRENT_SOURCE_DIR}/*.png")
+	#Shouldn't use this trick under Windows
```
Comment (tresf) [inline plugins/vestige/CMakeLists.txt]: This comment is implied as well.  Can be removed. :+1:

**Comment by tresf** [comment]
WTF is this for 
[Review on Reviewable](https://reviewable.io/reviews/lmms/lmms/2632)

**Comment by tresf** [comment]
Once the concerns are addressed, please squash your commits down into one and then we can merge. :+1:

**Comment by liushuyu** [comment]
>Once the concerns are addressed, please squash your commits down into one and then we can merge.

So I need to open a new PR since GitHub can't process `git rebase` and force push under PR condition.

**Comment by liushuyu** [comment]
Okay, I will do this now.

**Comment by tresf** [comment]
> So I need to open a new PR since GitHub can't process git rebase and force push under PR condition.

Yes it can.

**Comment by liushuyu** [comment]
It seems that Travis CI is pretty busy at this (very?) moment... Why not having another cup of tea (cake) ?

**Comment by tresf** [comment]
@liushuyu now you have a fast-forward commit in there.  Please squash that down too.

**Comment by liushuyu** [comment]
>@liushuyu now you have a fast-forward commit in there. Please squash that down too.

Sorry, I haven't encountered this scenario... And I think this PR is in a mess now, I suppose opening a new one can solve this ...?

**Comment by tresf** [comment]
@liushuyu good work. :)

**Comment by liushuyu** [comment]
Okay,  test passed... 🌝

**Pull request merged**